cpp/src/table/row_operators.cu

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/detail/structs/utilities.hpp>
+#include <cudf/detail/utilities/vector_factories.hpp>
+#include <cudf/table/experimental/row_operators.cuh>
+#include <cudf/table/table_view.hpp>
+
+#include <jit/type.hpp>
+
+namespace cudf {
+namespace experimental {
+
+namespace {
+
+auto struct_lex_verticalize(table_view table,
```

Comment:
This should probably be in the `lex` namespace instead of having `lex` in the name.

I kind of like "linearize" better than "verticalize". If a struct column is a tree with N leaves, then this function "linearizes" the tree into N "linear trees" (branch factor == 1).